services/bitbucket/bitbucket-pull-request.service.js

```diff
@@ -29,23 +29,23 @@ function pullRequestClassGenerator(raw) {
     static category = 'issue-tracking'
     static route = {
       base: `bitbucket/${routePrefix}`,
-      pattern: `:user/:repo`,
+      pattern: `:project/:repo`,
```

Comment:
While I can understand the thinking behind changing these from the Bitbucket Server/DC lens, I'm mildly opposed to changing it now given that we ended up going with a shared class and route to handle both Bitbucket Cloud and Server/DC.

There's other places where there's multiple viable nouns, but we always end up having to pick one (e.g. `:user` vs `:organization` for GitHub). At the end of the day the actual name doesn't matter all that much so long as it's not something outlandish, like `:somethingOutlandish` because our service is targeted at developers/technologists and we generally get to assume that folks can connect the minor dots like these.

It's not so much that I care about the actual value of the name, but want to avoid getting into a place where we have folks pushing to change these back and forth between the cloud/SaaS and self-hosted/on-prem version of tools like these.

Reply:
> It's not so much that I care about the actual value of the name, but want to avoid getting into a place where we have folks pushing to change these back and forth between the cloud/SaaS and self-hosted/on-prem version of tools like these.

I think the thing that was confusing about the existing setup is that in BB Server, there are users, but users can't own projects, so it's the project that owns the repo.

I agree that we should not just switch from the cloud/SaaS to the self-hosted/on-prem names, though wonder if it would make sense to parameterize the shared code so it can generate different routes for the cloud/SaaS and self-hosted/on-prem, particularly if that can be done without too much hassle, with the goal of making things clearer to users (in the frontend). Though I appreciate what you're saying about hoping folks can connect the minor dots.

Reply:
I guess if you agree @calebcartwright I'd be happy to take a stab at that (or to work with @danielle on same).

Reply:
we're discussing offline now actually, considering splitting the difference with a multiword name (e.g. `:projectOrUser`) since the former would better resonate with Server/DC while the latter more so with Cloud/SaaS.

I can't think of an example off hand but feel like we've done something similar in other routes before

Reply:
> I guess if you agree @calebcartwright I'd be happy to take a stab at that (or to work with @danielle on same).

sorry missed this below the other comment. i don't have any objections to providing more contextual names where we can, especially since it's not uncommon for their to be divergence between self-hosted and SaaS tool offerings in cases, just want to avoid having several extra class definitions where we can.

The other prominent case (though not a particularly addressable one) is on GitHub itself where a repo is owned by a user or an organization and we always use `:user`

Reply:
> The other prominent case (though not a particularly addressable one) is on GitHub itself where a repo is owned by a user or an organization and we always use `:user`

The thing I could imagine doing there is adding a description that can appear in the UI in these rare cases.

Reply:
Even on Cloud,  Projects seem to be visible these days though I find it a bit puzzling. What we need for our API call is still the owner/user/account and the repo name (AFAICT), not really sure how common it still is for default/unnanmed workspace and projects or the degree of possible confusion for Cloud users :man_shrugging:  

https://img.shields.io/bitbucket/pr-raw/guyzmo/python-bitbucket
![](https://img.shields.io/bitbucket/pr-raw/guyzmo/python-bitbucket)

https://bitbucket.org/guyzmo/python-bitbucket
![image](https://user-images.githubusercontent.com/13042488/105110701-c6a7ca00-5a84-11eb-8597-9d788fb0b819.png)

Reply:
@danielle did you see this discussion?

Reply:
Yes, are you stating that it should not be a compound name such as "projectOrUser" as requested but instead keep it as "project"?  My original pull request included "project" which matched the API documentation but I wasn't aware that there was a separate call for Bitbucket vs Bitbucket Server.

Reply:
I'm backtracking on my original thinking now. I feel like anything that indicates "project" as an option would cause issues for Cloud users, as we need the "owner" and the "repo" name, but if a Cloud user saw `:projectOrUser` and tried to use their Cloud project name the badge would not work.

I think the best option, if possible, would be to find a clean way to present different param names for the two routes to account for Server vs Cloud differences as Paul suggested in https://github.com/badges/shields/pull/6076#discussion_r560587197

Otherwise I think we'd be better off to try to find a generic singular term, and then just add help text that shows in the badge builder window

Reply:
So is the idea that we use `project` for Server and `user` for Cloud? (Or is it `owner` for Cloud?)

Reply:
In my experience `project` is what will most clearly resonate for `Server`. I've never been a user of BB Cloud so hard to say, though may just be worth sticking with `user` since that's what we have there now